src/spatial_reaction_systems/spatial_reactions.jl

```diff
@@ -0,0 +1,96 @@
+### Spatial Reaction Structures ###
+
+# Abstract spatial reaction structures.
+abstract type AbstractSpatialReaction end
+
+### EdgeParameter Metadata ###
+
+# Implements the edgeparameter metadata field.
+struct EdgeParameter end
+Symbolics.option_to_metadata_type(::Val{:edgeparameter}) = EdgeParameter
+
+# Implements the isedgeparameter check function.
+isedgeparameter(x::Num, args...) = isedgeparameter(Symbolics.unwrap(x), args...)
+function isedgeparameter(x, default = false)
+    p = Symbolics.getparent(x, nothing)
+    p === nothing || (x = p)
+    Symbolics.getmetadata(x, EdgeParameter, default)
+end
+
+### Transport Reaction Structures ###
+
+# A transport reaction. These are simple to hanlde, and should cover most types of spatial reactions.
+# Only permit constant rates (possibly consisting of several parameters).
+struct TransportReaction <: AbstractSpatialReaction
+    """The rate function (excluding mass action terms). Currently only constants supported"""
+    rate::Num
+    """The species that is subject to difusion."""
+    species::BasicSymbolic{Real}
+
+    # Creates a diffusion reaction.
+    function TransportReaction(rate::Num, species::Num)
+        new(rate, species.val)
+    end
+end
+# If the rate is a value, covert that to a numemric expression.
+function TransportReaction(rate::Number, args...)
+    TransportReaction(Num(rate), args...)
+end
+# Creates a vector of TransportReactions.
+function transport_reactions(transport_reactions)
+    [TransportReaction(tr[1], tr[2]) for tr in transport_reactions]
+end
+
+# Macro for creating a transport reaction.
+macro transport_reaction(rateex::ExprValues, species::Symbol)
+    make_transport_reaction(MacroTools.striplines(rateex), species)
+end
+function make_transport_reaction(rateex, species)
+    parameters = [];
+    find_parameters_in_rate!(parameters, rateex)
+    quote
+        $(isempty(parameters) ? nothing : :(@parameters $(parameters...)))
+        @variables t
+        @species $(species)(t)
+        TransportReaction($rateex, $species)
+    end
+end
+
+# Gets the parameters in a transport reaction.
+ModelingToolkit.parameters(tr::TransportReaction) = convert(Vector{BasicSymbolic{Real}}, Symbolics.get_variables(tr.rate))
+
+# Gets the species in a transport reaction.
+species(tr::TransportReaction) = [tr.species]
+spatial_species(tr::TransportReaction) = [tr.species]
```

Comment:
I think this was basically trying to be future-compatible. Basically, for `TransportReaction`s, simply having `spatial_species` to return the scalar would make sense. However, if we allow more complicated spatial reactions
1) There might be species that are not spatial species (e.g. transporter proteins that facilitate the transportation event, but are not transported themselves).
2) There may be more than 1 spatial species, which makes the vector make sense.

I guess `species` is not needed now, but I am a little bit worried that `spatial_species` may cause a problem down the line if we make it just
```julia
spatial_species(tr::TransportReaction) = tr.species
```

This considered, I think your call is better than mine, so happy to make the changes you still think is better.

Reply:
If you want to keep it a vector that is ok, as long as we then just use the field internally to avoid unnecessary allocations. I'll try to remember about this when going through the codegen parts. 

I think we can worry about future additions when we get to them. Let's just include what we need right now -- I think we have a pretty good and extensible design we've converged on we can add to later as needed.

Reply:
Sounds good. Will remove `species(tr::TransportReaction) = [tr.species]` and keep the other a vector. Will go through the internals and update so the field is used instead.

Would it make sense to define a field `spatial_species` in `TransportReaction` (and later on, similar for other `AbstractSpatialReaction`s) which contain this vector. Then this would return that vector (and we do not need to allocate when using this function).

Reply:
fixed.